components/server/ee/src/billing/billing-mode.ts

```diff
@@ -0,0 +1,121 @@
+/**
+ * Copyright (c) 2022 Gitpod GmbH. All rights reserved.
+ * Licensed under the GNU Affero General Public License (AGPL).
+ * See License-AGPL.txt in the project root for license information.
+ */
+
+import { inject, injectable } from "inversify";
+
+import { Team, User } from "@gitpod/gitpod-protocol";
+import { getExperimentsClientForBackend } from "@gitpod/gitpod-protocol/lib/experiments/configcat-server";
+import { SubscriptionService } from "@gitpod/gitpod-payment-endpoint/lib/accounting";
+import { Subscription } from "@gitpod/gitpod-protocol/lib/accounting-protocol";
+import { Config } from "../../../src/config";
+import { StripeService } from "../user/stripe-service";
+import { BillingMode } from "@gitpod/gitpod-protocol/lib/billing-modes";
+import { TeamSubscription2DB } from "@gitpod/gitpod-db/lib";
+
+@injectable()
+export class BillingModesImpl {
+    @inject(Config) protected readonly config: Config;
+    @inject(SubscriptionService) protected readonly subscriptionSvc: SubscriptionService;
+    @inject(StripeService) protected readonly stripeSvc: StripeService;
+    @inject(TeamSubscription2DB) protected readonly teamSubscription2Db: TeamSubscription2DB;
+
+    async getBillingModeForUser(user: User): Promise<BillingMode> {
+        if (!this.config.enablePayment) {
+            // Payment is not enabled. E.g. Self-Hosted.
+            return { mode: "none" };
+        }
+        const now = new Date();
+
+        // Is Usage Based Billing enabled for this user or not?
+        const isUsageBasedBillingEnabled = await getExperimentsClientForBackend().getValueAsync(
+            "isUsageBasedBillingEnabled",
+            false,
+            {
+                user,
+            },
+        );
```

Comment:
Wait, isn't UBB flag controlled by whether you're part of the Gitpod team or not? In this case I believe you may need to also pass along the `teams` of the user.

Reply:
Yes, that's up to discussion. See [comment](https://github.com/gitpod-io/gitpod/pull/11601#discussion_r934295916) below.